libr/bin/bfile.c

```diff
@@ -875,37 +874,42 @@ R_API int r_bin_file_hash(RBin *bin, ut64 limit, const char *file) {
 	buf_len = r_io_desc_size (iod);
 	// By SLURP_LIMIT normally cannot compute ...
 	if (buf_len > limit) {
-		eprintf ("Cannot compute hash\n");
-		return -1;
-	}
-	//  XXX should use io api not raw file slurping
-	int blen = 0;
-	buf = (ut8*)r_file_slurp (file, &blen);
-	if (!buf) {
+		eprintf ("Cannot compute hash: %zu/%zu\n", buf_len, limit);
 		return false;
 	}
-	buf_len = blen;
-	if (!o->info) {
+#define BLK_SIZE_OFF 64000
```

Comment:
better use a `const size_t blocksize = 64000;` no need to use a define which its scope is global